# Preview of PTB-XL Statements

## 1. Import libraries

In [1]:
import pandas as pd

## 2. Set file path

In [2]:
labels_path = "../data/ptb-xl-plus/ptbxl_statements.csv"

## 3. Load the statements dataset

In [4]:
df_labels = pd.read_csv(labels_path)
df_labels.head()

,ecg_id,scp_codes,scp_codes_ext,scp_codes_ext_snomed
0,1,"[('NORM', 100.0), ('LVOLT', 100.0), ('SR', 100...","[('NORM', 100.0), ('LVOLT', 100.0), ('SR', 100...","[(320536, 100.0), (441840, 100.0), (4008453, 1..."
1,2,"[('NORM', 80.0), ('SBRAD', 100.0)]","[('NORM', 80.0), ('SBRAD', 100.0)]","[(134057, 100.0), (321588, 100.0), (440142, 10..."
2,3,"[('NORM', 100.0), ('SR', 100.0)]","[('NORM', 100.0), ('SR', 100.0)]","[(441840, 100.0), (4008453, 100.0), (4010019, ..."
3,4,"[('NORM', 100.0), ('SR', 100.0)]","[('NORM', 100.0), ('SR', 100.0)]","[(441840, 100.0), (4008453, 100.0), (4010019, ..."
4,5,"[('NORM', 100.0), ('SR', 100.0)]","[('NORM', 100.0), ('SR', 100.0)]","[(441840, 100.0), (4008453, 100.0), (4010019, ..."


## 4. Data shape


In [5]:
df_labels.shape

(21799, 4)

In [7]:
df_labels.columns.tolist()

['ecg_id', 'scp_codes', 'scp_codes_ext', 'scp_codes_ext_snomed']

The file contains four columns:

- `ecg_id` - ECG record identifier, used to join labels with features.
- `scp_codes` - original diagnostic codes for the ECG record.
- `scp_codes_ext` - extended version of the diagnostic codes.
- `scp_codes_ext_snomed` - diagnostic codes mapped to SNOMED CT medical codes.

(SNOMED CT codes are standard medical codes used to describe diagnoses in the same way across different healthcare systems.)

Example interpretation:

- `ecg_id = 1` - first ECG record in the dataset.
- `NORM, 100.0` - the ECG is described as normal with certainty 100.
- `LVOLT, 100.0` - low voltage is detected with certainty 100.
- `SR, 100.0` - sinus rhythm is detected with certainty 100.

SNOMED CT (`scp_codes_ext_snomed`) - For example, `(441840, 100.0)` means that the ECG record has a specific medical label with certainty 100. 

## 5. Check data types

In [8]:
df_labels.info()

<class 'pandas.DataFrame'>
RangeIndex: 21799 entries, 0 to 21798
Data columns (total 4 columns):
 #   Column                Non-Null Count  Dtype
---  ------                --------------  -----
 0   ecg_id                21799 non-null  int64
 1   scp_codes             21799 non-null  str  
 2   scp_codes_ext         21799 non-null  str  
 3   scp_codes_ext_snomed  21799 non-null  str  
dtypes: int64(1), str(3)
memory usage: 15.6 MB


## 6. Missing values analysis

In [9]:
df_labels.isna().sum().sort_values(ascending=False).head(30)

ecg_id                  0
scp_codes               0
scp_codes_ext           0
scp_codes_ext_snomed    0
dtype: int64

There are no missing values in this file.  
All label columns are complete and can be used in the next steps.